x-pack/plugins/ml/public/application/datavisualizer/data_drift/data_drift_index_patterns_editor.tsx

```diff
@@ -0,0 +1,416 @@
+
+    getMatchingDataView();
+
+    return () => {
+      unmounted = true;
+    };
+  }, [referenceIndexPattern, productionIndexPattern, timeField, dataViews]);
+  const createDataViewAndRedirectToDataDriftPage = async (createAdHocDV = false) => {
+    // Create adhoc data view
+    const indicesName = getDefaultIndexPattern(referenceIndexPattern, productionIndexPattern);
+
+    const timeFieldName =
+      Array.isArray(timeField) && timeField.length > 0 ? timeField[0].value : undefined;
+
+    let dataView;
+
+    if (!foundDataViewId) {
+      const defaultDataViewName =
+        dataViewMsg === undefined
+          ? indicesName
+          : `${indicesName}${timeFieldName ? '-' + timeFieldName : ''}`;
+
+      const modifiedDataViewName = dataViewName === '' ? defaultDataViewName : dataViewName;
+      if (canEditDataView && createAdHocDV === false) {
+        dataView = await dataViews.createAndSave({
+          title: indicesName,
+          name: modifiedDataViewName,
+          timeFieldName,
+        });
+      } else {
+        dataView = await dataViews.create({
+          title: indicesName,
+          name: modifiedDataViewName,
+          timeFieldName,
+        });
+      }
+    }
+    const dataViewId = foundDataViewId ?? dataView?.id;
+    const url = await locator.getUrl({
+      page: ML_PAGES.DATA_DRIFT,
+      pageState: {
+        index: dataViewId,
+        reference: referenceIndexPattern,
+        production: productionIndexPattern,
+        timeFieldName,
+      },
+    });
+
+    await navigateToPath(url);
+  };
+
+  const hasError =
+    refError !== undefined ||
+    comparisonError !== undefined ||
+    !productionIndexPattern ||
+    !referenceIndexPattern;
+
+  const firstSetOfSteps = [
+    {
+      title: 'Pick index pattern for reference data',
+      children: (
+        <EuiFlexItem grow={false}>
+          <DataViewEditor
+            key={'reference'}
+            label={
+              <FormattedMessage
+                id="xpack.ml.dataDrift.indexPatternsEditor.referenceData"
+                defaultMessage="Select an index pattern to specify the reference data."
+              />
+            }
+            helpText={
+              <FormattedMessage
+                id="xpack.ml.dataDrift.indexPatternsEditor.requireSameTimeFieldHelpText"
+                defaultMessage="To view the changes in the data over time, reference and comparison data can have different index patterns, but they must have the same time field."
+              />
+            }
+            dataViewEditorService={referenceDataViewEditorService}
+            indexPattern={referenceIndexPattern}
+            setIndexPattern={setReferenceIndexPattern}
+            onError={setRefError}
+          />
+        </EuiFlexItem>
+      ),
+    },
+    {
+      title: 'Pick index pattern for comparison data',
+      children: (
+        <EuiFlexItem grow={false}>
+          <DataViewEditor
+            key={'comparison'}
+            label={
```

Comment:
As above, is this label needed at all? Or just be `Index pattern`? 'Select' is a bit misleading as you can't select an item from the list in the UI.

<img width="549" alt="image" src="https://github.com/elastic/kibana/assets/7405507/13d17d36-0bd2-4dff-8ee3-20d39012db69">

Any thoughts @szabosteve ?

Reply:
As above.

Reply:
Updated here [`f2f5a38` (#162853)](https://github.com/elastic/kibana/pull/162853/commits/f2f5a385cb03e8aaf0b41ee595c410193686928f)